### PR TITLE
Doc typos fixed

### PR DESCRIPTION
**Pull request by nbro**
This pull request

- fixes several typos in some docstrings
- adds the missing documentation of a parameter of SearchConfiguration
- updates the README because vector types are now supported

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -76,3 +76,7 @@ venv.bak/
 
 # Editor Files
 .idea
+
+examples/santa*
+examples/artificial*
+examples/final_individual
```
Comment (erp12) [inline .gitignore]: I'm guessing these are some examples that you added in your fork/clone, but don't want to contribute to the main repo (at least not yet 😃 ). If that is correct, then this `.gitignore` change probably makes sense for your fork, but not the main repository.  If not, then can you explain this change a bit more.

**Review by erp12 — CHANGES_REQUESTED** [review]
Thank so much for doing this! Sorry for taking so long to get around for a review. 
Overall, these fixes are great! I only have 1 minor comment.

File: .gitignore (hunk 1 above)
Comment (nbro) [reply .gitignore]: @erp12 You're right: I was working on an example, which was not yet finished. 

What do you suggest to do now? Should I perform another commit that deletes those lines in the `.gitignore`?

Comment (erp12) [reply .gitignore]: Yup, that would be great! Thanks.

Comment (nbro) [reply .gitignore]: @erp12 Ok. I've done that. Check it.

**Comment by erp12** [comment]
Sorry for taking forever to get around to merging this. Looks great, thanks so much for taking the time to make pyshgp better!

**Pull request merged**